content/TruenasApps/UsingCustomApp.md

```diff
@@ -48,138 +35,1199 @@ If you are unsure about any configuration settings, review the [Install Custom A
-   Enter the **Mount Path** to mount the host path inside the container.
+   See [Setting up Storage](#setting-up-app-storage) above for more information.
 
    {{< trueimage src="/images/SCALE/Apps/InstallCustomAppScreenStorage.png" alt="Storage Settings" id="Storage Settings" >}}
 
-   Add any memory-backed or other volumes you need or want to use.
-   You can add more volumes to the container later, if needed.
+   Click **Add** to display a block of configuration settings for each storage mount.
+   You can edit to add additional storage to the container later, if needed.
+
+   Use the dropdown to select the storage **Type** you need to mount then configure the appropriate settings.
+
+   {{< expand "ixVolume (Dataset created automatically by the system)" "v" >}}
+   To allow TrueNAS to create the storage volume, leave **Type** set to **ixVolume (Dataset created automatically by the system)**.
+   This is recommended for a test deployment of an app but not for a full app deployment.
+
+   {{< trueimage src="/images/SCALE/Apps/InstallCustomAppAddixVolume.png" alt="ixVolume Settings" id="ixVolume Settings" >}}
+
+   Enter in **Mount Path** the <file>**path/to/directory**</file> where the ixVolume mounts inside the container.
+
+   Select whether to mount the ixVolume as **Read Only** or to **Enable ACL** and configure settings as needed.
+   {{< /expand >}}
+
+   {{< expand "Host Path (Path that already exists on the system)" "v" >}}
+   To configure a persistent host path to an existing dataset, select **Host Path (Path that already exists on the system)**.
+   This is the recommended setting for on-disk storage in a production deployment.
 
-8. Enter any additional settings required for your application, such as workload details or adding container settings for your application.
+   {{< trueimage src="/images/SCALE/Apps/InstallCustomAppAddHostPathVol.png" alt="Host Path Settings" id="Host Path Settings" >}}
 
-9. Select the **Scaling/Upgrade Policy** to use.
-   The default is **Kill existing pods before creating new ones**.
+   Enter in **Mount Path** the <file>**path/to/directory**</file> where the host path mounts inside the container.
+   Then define the **Host Path**. Enter a path or click <span class="material-icons">arrow_right</span> to the left of <svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24"><path fill="currentColor" d="M5 21L3 9h18l-2 12zm5-6h4q.425 0 .713-.288T15 14t-.288-.712T14 13h-4q-.425 0-.712.288T9 14t.288.713T10 15M6 8q-.425 0-.712-.288T5 7t.288-.712T6 6h12q.425 0 .713.288T19 7t-.288.713T18 8zm2-3q-.425 0-.712-.288T7 4t.288-.712T8 3h8q.425 0 .713.288T17 4t-.288.713T16 5z"/></svg> **/mnt** to browse to the location of the dataset in TrueNAS.
 
-10. Use **Resource Reservation** to allocate GPU resources if available and required for the application.
+   Select whether to mount the host path as **Read Only** or to **Enable ACL** and configure settings as needed.
```

Comment:
in most of the apps, you need to enable the acl or the app does not deploy, so is that the case here as well? Also, you could use the snippets created for apps tutorials for this content.

Reply:
My understanding of it is you need to set permissions correctly, but that could be directly on the dataset or by enabling ACL here? 

And this is one of the cases where the snippet doesn't work, because the ACL snippet includes the sentence "* The default apps user which is either **473** for apps in the stable train or **568** for apps in the **enterprise** or **community** train," which doesn't make sense in the Custom App process. 

I can bring over a bit more detail from the snippet but it would read awkwardly to just include it.

Reply:
On ACLs, I asked Stavros if you could use the dataset permissions widget to access the Edit ACL form and he said no, you must use the ACL option in the container through the install widget.
If Stavros is not imposing this ACL requirement in the custom apps wizard then we just need to be clear that it is an option but not a requirement.

Reply:
Checking with Stavros.